drop-in/src/main/java/com/adyen/checkout/dropin/internal/ui/DropInViewModelFactory.kt

```diff
@@ -35,7 +37,9 @@ internal class DropInViewModelFactory(
     override fun <T : ViewModel> create(key: String, modelClass: Class<T>, handle: SavedStateHandle): T {
         val bundleHandler = DropInSavedStateHandleContainer(handle)
 
-        val dropInConfiguration: DropInConfiguration = requireNotNull(bundleHandler.dropInConfiguration)
+        val dropInConfiguration: DropInConfiguration = requireNotNull(bundleHandler.dropInConfiguration).apply {
+            bundleHandler.overridePaymentMethodInformation(overriddenPaymentMethodInformation)
```

Comment:
Also nitpicking: I like to use `apply` to modify or execute something in the context object itself (`dropInConfiguration` in this case). While here we're using it to read `dropInConfiguration.overriddenPaymentMethodInformation` and pass it on to modify the `paymentMethodsApiResponse` inside the `bundleHandler`. So first thing I assumed while reading this is that `bundleHandler` is a property inside `dropInConfiguration` and we're modifying it.

If you agree then I would suggest using `bundleHandler.dropInConfiguration?.let` or even just explicitly calling `dropInConfiguration.overriddenPaymentMethodInformation`, it does add more code but it also adds more clarity. But feel free to disagree, this is really just a personal preference that I thought I'd share.

Reply:
I agree that it might be confusing to why `.apply` was used here. I will modify it.